spec/02-integration/11-dbless/01-respawn_spec.lua

```diff
@@ -0,0 +1,73 @@
+local helpers = require "spec.helpers"
+
+describe("worker respawn", function()
+  local admin_client, proxy_client
+
+  lazy_setup(function()
+    assert(helpers.start_kong({
+      database   = "off",
+    }))
+  end)
+
+  lazy_teardown(function()
+    helpers.stop_kong(nil, true)
+  end)
+
+  before_each(function()
+    admin_client = assert(helpers.admin_client())
+    proxy_client = assert(helpers.proxy_client())
+  end)
+
+  after_each(function()
+    if admin_client then
+      admin_client:close()
+    end
+
+    if proxy_client then
+      proxy_client:close()
+    end
+  end)
+
+  it("lands on the correct cache page #5799", function()
+    local res = assert(admin_client:send {
+      method = "POST",
+      path = "/config",
+      body = {
+        config = [[
+        _format_version: "1.1"
+        services:
+        - name: my-service
+          url: https://example.com
+          plugins:
+          - name: key-auth
+          routes:
+          - name: my-route
+            paths:
+            - /
+
+        consumers:
```

Comment:
This is not necessary, because when wrong cache page is used `404` will be returned due to wrong plugin iterator being built. But it won't hurt to test for `200` as well.